packages/generator/src/parser/swagger-parser.ts

```diff
@@ -13,9 +13,7 @@ export function parseSwaggerFromPath(swaggerPath: PathLike): SwaggerMetadata {
 
 function parseSwaggerFile(swaggerFile: string): SwaggerMetadata {
   const swaggerMetaData = JSON.parse(swaggerFile);
-  // If the file is not swagger but openapi
-  if (swaggerMetaData.openapi) {
-    swaggerMetaData.definitions = swaggerMetaData?.components?.schemas;
-  }
+  // Get definitions from schema, this is typically the case for openapi files
+  swaggerMetaData.definitions = swaggerMetaData.definitions || swaggerMetaData?.components?.schemas;
```

Comment:
The original concern was because this function was applied for both swagger and openapi (new case) file.
However, I checked the open api definition, the `components.schemas` only exist in version 3 not version 2.
Checking the openapi version (=3.x.y?) makes it safer, but you are right, nothing should happen if you apply to version 2 files.
But since we have openapi file for both version 2 and 3 in the `s4-api` folder, I would make it less confusing.
It can be achieved by adding a comment to this line (line 17) or old version (line 18), so we are aware this adjustment is only for version 3 not all versions.